scripts/train_linker.py

```diff
@@ -0,0 +1,72 @@
+from sklearn.ensemble import GradientBoostingClassifier
+from sklearn.metrics import accuracy_score, classification_report
+from joblib import dump, load
+import jsonlines
+import datetime
+from scispacy import umls_semantic_type_tree
+from linking import featurizer
+import argparse
+from tqdm import tqdm
+
+
+def main(data_path: str):
+    train = []
+    dev = []
+    test = []
+    root = ''
+    files = [f'{data_path}/train.jsonl', f'{data_path}/dev.jsonl', f'{data_path}/test.jsonl']
+    lists = [train, dev, test]
+    # limits = [40000000, 4000000, 4000000]
+    limits = [4000000, 0, 4000000]
+
+    start_time = datetime.datetime.now()
+
+    # load data from files
+    for filename, data, limit in zip(files, lists, limits):
```

Comment:
This whole block took me about 5 minutes to parse, when it should have taken me 30 seconds. 

1. You are looping over the data twice, once to read it and once to featurize it. 
2. It's not really necessary to create all these data lists beforehand, when you are just building a list of them anyway. why not just define `data = []` inside the loop body? It would make the code much easier to read because I don't have to go hunting to find what `data` is.

Reply:
is this better ?